src/hotspot/share/classfile/systemDictionary.hpp

```diff
@@ -81,6 +81,23 @@ class EventClassLoad;
 class Symbol;
 class TableStatistics;
 
+class SharedClassLoadingMark {
```

Comment:
I think it's better to put this class into systemDictionaryShared.hpp

Reply:
The move to systemDictionaryShared.hpp will cause zero build failed. We need a guard for CDS at:
1294     if (k != NULL) {
1295       SharedClassLoadingMark slm(THREAD, k);
1296       k = find_or_define_instance_class(class_name, class_loader, k, CHECK_NULL);
1297     }
That makes the code looks fragmented. Do you agree to keep it not moved?

Reply:
Maybe we should add guard here for shared code --- putting it to systemDictionaryShared.hpp is more reasonable.